packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -537,34 +537,12 @@ class IOSDevice extends Device {
       int installationResult = 1;
       if (debuggingOptions.debuggingEnabled) {
         _logger.printTrace('Debugging is enabled, connecting to vmService');
-        final DeviceLogReader deviceLogReader = getLogReader(
-          app: package,
-          usingCISystem: debuggingOptions.usingCISystem,
-        );
-
-        // If the device supports syslog reading, prefer launching the app without
-        // attaching the debugger to avoid the overhead of the unnecessary extra running process.
-        if (majorSdkVersion >= IOSDeviceLogReader.minimumUniversalLoggingSdkVersion) {
-          iosDeployDebugger = _iosDeploy.prepareDebuggerForLaunch(
-            deviceId: id,
-            bundlePath: bundle.path,
-            appDeltaDirectory: package.appDeltaDirectory,
-            launchArguments: launchArguments,
-            interfaceType: connectionInterface,
-            uninstallFirst: debuggingOptions.uninstallFirst,
-          );
-          if (deviceLogReader is IOSDeviceLogReader) {
-            deviceLogReader.debuggerStream = iosDeployDebugger;
-          }
-        }
-        // Don't port foward if debugging with a wireless device.
-        vmServiceDiscovery = ProtocolDiscovery.vmService(
-          deviceLogReader,
-          portForwarder: isWirelesslyConnected ? null : portForwarder,
-          hostPort: debuggingOptions.hostVmServicePort,
-          devicePort: debuggingOptions.deviceVmServicePort,
+        vmServiceDiscovery = _setupDebuggerAndVmServiceDiscovery(
```

Comment:
from a cursory look, this diff doesn't change anything, just factors out the left side logic to a helper method, is that right?

Reply:
Pretty much, only difference is I added a `skipInstall` option